mreg_cli/api/models.py

```diff
@@ -2502,17 +2500,16 @@ class Host(FrozenModelWithTimestamps, WithTTL, WithHistory, APIMixin):
 
     @field_validator("name", mode="before")
     @classmethod
-    def validate_name(cls, value: str) -> HostT:
+    def validate_name(cls, value: Any) -> HostT:
         """Validate the hostname."""
         return HostT(hostname=value)
 
     @field_validator("bacnetid", mode="before")
     @classmethod
-    def convert_bacnetid(cls, v: dict[str, int] | None) -> int | None:
-        """Convert json id field to int or None."""
-        if v and "id" in v:
-            return v["id"]
-
+    def convert_bacnetid(cls, v: Any) -> Any:
+        """Use nested ID value in bacnetid value."""
+        if isinstance(v, dict):
+            return v.get("id")  # pyright: ignore[reportUnknownVariableType, reportUnknownMemberType]
```

Comment:
Not guaranteed to be a dict, check first. The validation of the value we retrieve is left to Pydantic.